build.rs

```diff
@@ -110,19 +129,57 @@ fn need_gn_ninja_download() -> bool {
     && env::var_os("GN").is_some())
 }
 
+fn is_compatible_clang_version(clang_path: &Path) -> bool {
```

Comment:
Cool!

Reply:
We'd just need to update the compatible versions constants roughly once per year during a V8 upgrade once Google decides to break compatibility with older `clang`'s again.

Another side node: It looks like a lot of linux distributions ship their versioned `clang` packages with a version suffix (e.g. `clang-8`), which is unfortunately not compatible with Chromium's build scripts (and I guess we don't want to patch them too heavily). So this approve is only compatibly with: Apple clang (on macOS), homebrew's llvm packages (on macOS and linux via Homebrew on Linux?), the official llvm binaries (from http://releases.llvm.org/, all platforms) and maybe the default `clang` package on Linux (but they are often too old anyway).

Reply:
it might be worthwhile to add some of this in a comment